Lib/test/test_compileall.py

```diff
@@ -460,31 +460,18 @@ def test_error(self):
 class CommandLineTestsBase:
     """Test compileall's CLI."""
 
-    @classmethod
-    def setUpClass(cls):
-        for path in filter(os.path.isdir, sys.path):
-            directory_created = False
-            directory = pathlib.Path(path) / '__pycache__'
-            path = directory / 'test.try'
-            try:
-                if not directory.is_dir():
-                    directory.mkdir()
-                    directory_created = True
-                path.write_text('# for test_compileall', encoding="utf-8")
-            except OSError:
-                sys_path_writable = False
-                break
-            finally:
-                os_helper.unlink(str(path))
-                if directory_created:
-                    directory.rmdir()
-        else:
-            sys_path_writable = True
-        cls._sys_path_writable = sys_path_writable
-
-    def _skip_if_sys_path_not_writable(self):
-        if not self._sys_path_writable:
-            raise unittest.SkipTest('not all entries on sys.path are writable')
+    @contextlib.contextmanager
+    def _use_pycache_prefix(self, basedir):
```

Comment:
I suggest to a different name and omit the parameter which is always self.directory:

```suggestion
    def set_pycache_prefix(self):
```

Reply:
I was attempting to match the "style" of this file as best as possible.  I have now renamed it to match the context manager used within test_importlib.  Also, "set" doesn't feel right for use in a with statement.  I guess it comes from English grammar of us*ing* present participle (-ing forms) after prepositions (`with`).